src/ops.rs

```diff
@@ -0,0 +1,79 @@
+//! Additional (primarily arithmetic) operators for working with memory.
+
+/// A checked equivalent to the `+` operator. Returning an `Option` signifying
+/// if the operation result is within bounds.
+///
+/// Similarly to `std::ops::Add` note that `Rhs` is `Self` by default, but
+/// this is not mandatory.
```

Comment:
We should mention how this is different from both `std::ops::Add` and `num::CheckedAdd` while also including links. Maybe something like:
```rust
/// A checked equivalent to the `+` operator that returns `None` instead
/// of panicking or wrapping on overflow. Similar to [`std::ops::Add`]
/// an `Output` type _must_ be specified and an `Rhs` type _may_ be
/// specified (defaults to `Self`).
///
/// We define our own trait instead of using
/// [`num::CheckedAdd`](https://docs.rs/num/latest/num/trait.CheckedAdd.html)
/// because that trait essentially mandates that `Rhs` and `Output` be `Self`.
```